Core/Object Arts/Dolphin/Base/ArrayedCollection.cls

```diff
@@ -121,7 +121,36 @@ sortUsing: aSortAlgorithm
 	aSortAlgorithm 
 		sort: self
 		from: 1
-		to: self size! !
+		to: self size!
+
+storeOn: aStream indent: anInteger variables: aDictionary
+	"Append to the <puttableStream> argument, target, an expression which when 
+	evaluated will answer a collection similar to the receiver."
+
+	(self allSatisfy: [:each | each isImmutable]) ifTrue: [
```

Comment:
isImmutable is not limited to things with literal form. You'll need to use #isLiteral for this purpose. In fact this is already correctly defined on the relevant collections (Array and ByteArray)